From 271983b94021991b410b35516b316f6d64766b26 Mon Sep 17 00:00:00 2001
From: Amalendu Prasad Das <amalendu.prasad.das@intel.com>
Date: Fri, 26 Jul 2019 10:15:50 +0530
Subject: [PATCH] Bluetooth cannot reconnect after BT turn OFF/ON

Bluetooth profile proxy connection thread was blocked by recursive
function call, which try to take the Reentrant Lock recursively.
Due to which BluetoothBroadcastReceiver is not able to receive the
broadcast message

Change-Id: I194baef4e645309c6e3666e9f505c16da7dffce0
Tracked-On: https://jira.devtools.intel.com/browse/OAM-84277
Signed-off-by: Amalendu Prasad Das <amalendu.prasad.das@intel.com>
---
 service/src/com/android/car/CarBluetoothUserService.java | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/service/src/com/android/car/CarBluetoothUserService.java b/service/src/com/android/car/CarBluetoothUserService.java
index 60bc56179..ead930aa7 100644
--- a/service/src/com/android/car/CarBluetoothUserService.java
+++ b/service/src/com/android/car/CarBluetoothUserService.java
@@ -283,10 +283,6 @@ public class CarBluetoothUserService extends ICarBluetoothUserService.Stub {
         } finally {
             mBluetoothProxyStatusLock.unlock();
         }
-        if (!proxyConnected) {
-            setupBluetoothConnectionProxies();
-            return isBluetoothConnectionProxyAvailable(profile);
-        }
         return proxyConnected;
     }
 
-- 
2.17.1

